cam_diagnostics/tropopause_diagnostics.F90

```diff
@@ -0,0 +1,220 @@
+    call history_add_field('TROP_Z',   'tropopause_geopotential_height_wrt_surface', horiz_only,  'avg', 'm' )
+    call history_add_field('TROP_DZ',  'geopotential_height_difference_between_atmosphere_layer_and_tropopause', 'lev', 'avg', 'm')
+    call history_add_field('TROP_PD',  'probability_distribution_of_tropopause_vertical_layer_index', 'lev', 'avg', 'probability')
+    call history_add_field('TROP_FD',  'tropopause_found', horiz_only,  'avg', 'probability')
+
+    ! Primary (Lapse rate) only
+    call history_add_field('TROPP_P',  'tropopause_air_pressure_from_lapse_rate_method', horiz_only,  'avg', 'Pa')
+    call history_add_field('TROPP_T',  'tropopause_air_temperature_from_lapse_rate_method', horiz_only,  'avg', 'K' )
+    call history_add_field('TROPP_Z',  'tropopause_geopotential_height_wrt_surface_from_lapse_rate_method', horiz_only, 'avg', 'm' )
+    call history_add_field('TROPP_DZ', 'geopotential_height_difference_between_atmosphere_layer_and_tropopause_from_lapse_rate_method', 'lev', 'avg', 'm')
+    call history_add_field('TROPP_PD', 'probability_distribution_of_tropopause_vertical_layer_index_from_lapse_rate_method', 'lev', 'avg', 'probability')
+    call history_add_field('TROPP_FD', 'tropopause_found_from_lapse_rate_method', horiz_only,  'avg', 'probability')
+
+    ! Cold point (CPP) only
+    call history_add_field('TROPF_P',  'tropopause_air_pressure_from_cold_point_method', horiz_only,  'avg', 'Pa')
+    call history_add_field('TROPF_T',  'tropopause_air_temperature_from_cold_point_method', horiz_only,  'avg', 'K' )
+    call history_add_field('TROPF_Z',  'tropopause_geopotential_height_wrt_surface_from_cold_point_method', horiz_only,  'avg', 'm' )
+    call history_add_field('TROPF_DZ', 'geopotential_height_difference_between_atmosphere_layer_and_tropopause_from_cold_point_method', 'lev', 'avg', 'm')
+    call history_add_field('TROPF_PD', 'probability_distribution_of_tropopause_vertical_layer_index_from_cold_point_method', 'lev', 'avg', 'probability')
+    call history_add_field('TROPF_FD', 'tropopause_found_from_cold_point_method', horiz_only,  'avg', 'probability')
+
+    ! Climatology only - will never fail
+    call history_add_field('TROPC_P',  'tropopause_air_pressure_from_climatological_method', horiz_only,  'avg', 'Pa')
+    call history_add_field('TROPC_T',  'tropopause_air_temperature_from_climatological_method', horiz_only,  'avg', 'K' )
+    call history_add_field('TROPC_Z',  'tropopause_geopotential_height_wrt_surface_from_climatological_method', horiz_only, 'avg', 'm' )
+    call history_add_field('TROPC_DZ', 'geopotential_height_difference_between_atmosphere_layer_and_tropopause_from_climatological_method', 'lev', 'avg', 'm')
+    call history_add_field('TROPC_PD', 'probability_distribution_of_tropopause_vertical_layer_index_from_climatological_method', 'lev', 'avg', 'probability')
+    call history_add_field('TROPC_FD', 'tropopause_found_from_climatological_method', horiz_only,  'avg', 'probability')
```

Comment:
Thank you for the great question! I scanned throughout the CAM code and these are the methods that are used in the current version of the code. Others in CAM were put behind a logical to output all methods, but it was hardcoded in the `tropopause.F90` code as `.false.`. Because of the (a little unfortunate) duplication of code / standard names when adding a new method I supported only the ones that are currently used, and we have a cleanup issue (#116) to remove the unused methods eventually.